core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,19 +135,33 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var deleteEmptyElements = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            deleteEmptyElements();
             $scope.isAddItemButtonPresent = true;
           };
 
           $scope.hideAddItemButton = function() {
             $scope.isAddItemButtonPresent = false;
           };
 
+          if ($scope.localValue.length === 0) {
```

Comment:
Actually, @kirankonduru -- I just realized that this should not go here. It forces pretty much every list on the entire site to be nonempty, but in places like the admin page we don't want that, right?

Better to have this done in SetInput instead.